src/Morphir/JsonSchema/Backend.elm

```diff
@@ -105,55 +105,96 @@ extractTypes modName definition =
+            case ( FQName.toString fQName, argTypes ) of
+                ( "Morphir.SDK:Basics:int", [] ) ->
+                    Integer
 
-        Type.CustomTypeDefinition typeArgs accessControlledConstructors ->
-            Nothing
+                ( "Morphir.SDK:Decimal:decimal", [] ) ->
+                    String
 
+                ( "Morphir.SDK:String:string", [] ) ->
+                    String
 
-mapType : Type ta -> Maybe SchemaType
-mapType typ =
-    case typ of
-        Type.Reference a fQName types ->
-            case FQName.toString fQName of
-                "Morphir.SDK:Basics:int" ->
-                    Just Integer
+                ( "Morphir.SDK:Char:char", [] ) ->
+                    String
 
-                "Morphir.SDK:String:string" ->
-                    Just String
+                ( "Morphir.SDK:Basics:float", [] ) ->
+                    Number
 
-                "Morphir.SDK:Basics:float" ->
-                    Just Number
+                ( "Morphir.SDK:Tuple:tuple", [] ) ->
+                    Number
 
-                "Morphir.SDK:Basics:bool" ->
-                    Just Boolean
+                ( "Morphir.SDK:Basics:bool", [] ) ->
+                    Boolean
 
-                "Morphir.SDK:List:list" ->
-                    List.head types
-                        |> Maybe.andThen mapType
-                        |> Maybe.map Array
+                ( "Morphir.SDK:List:list", [ itemType ] ) ->
+                    Array (ListType (mapType itemType)) False
 
-                _ ->
-                    Nothing
+                ( "Morphir.SDK:Set:set", [ itemType ] ) ->
+                    Array (ListType (mapType itemType)) True
+
+                ( "Morphir.SDK:Maybe:maybe", [ itemType ] ) ->
+                    OneOf [ Null, mapType itemType ]
 
-        Type.Record a fields ->
+                _ ->
+                    Ref
+                        (String.concat
+                            [ "#/$defs/"
+                            , moduleName |> Path.toString Name.toTitleCase "."
+                            , "."
+                            , localName |> Name.toTitleCase
+                            ]
+                        )
+
+        Type.Record _ fields ->
             fields
                 |> List.foldl
-                    (\field maybeDictSoFar ->
-                        maybeDictSoFar
-                            |> Maybe.map2
-                                (\schemaType dictSofar ->
-                                    Dict.insert (Name.toTitleCase field.name) schemaType dictSofar
-                                )
-                                (mapType field.tpe)
+                    (\field ->
+                        \dictSofar ->
+                            Dict.insert (Name.toTitleCase field.name) (mapType field.tpe) dictSofar
                     )
-                    (Just Dict.empty)
-                |> Maybe.map Object
+                    Dict.empty
```

Comment:
Using a `foldl` here adds unnecessary complexity in my opinion. You could replace it with a `map`.